### PR TITLE
Crash fix, parseTypeName can return null.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/parsing/Parser.cpp [hunk 1]
```diff
@@ -1244,7 +1244,8 @@ ASTPointer<Expression> Parser::parseLeftHandSideExpression(
 	{
 		expectToken(Token::New);
 		ASTPointer<TypeName> typeName(parseTypeName(false));
-		nodeFactory.setEndPositionFromNode(typeName);
+		if (typeName)
+			nodeFactory.setEndPositionFromNode(typeName);
```
Comment (axic) [inline libsolidity/parsing/Parser.cpp:1248]: `else nodeFactory.markEndPosition()` ?

Comment (chriseth) [reply libsolidity/parsing/Parser.cpp:1248]: it marks some end position automatically

Comment (chriseth) [reply libsolidity/parsing/Parser.cpp:1248]: ah sorry, I'm wrong there

**Comment by chriseth** [comment]
Fixed.

**Review by axic — APPROVED** [review]

**Pull request merged**